src/clients/PuchasesClient.ts

```diff
@@ -0,0 +1,86 @@
+import { IPurchasesClient } from "../interfaces/purchases.interfaces";
+import { ApiCustomerPurchases, ApiCustomerSummary, ClientOptions, GetCustomerPurchasesCallback, GetCustomersBulkInfoCallback, GetCustomersBulkInfoParams, Paginated } from "../types";
+import { BaseClient } from "./BaseClient.abstract";
+
+/**
+ * class PurchasesClient
+ * constructor requires options as
+ * {
+ *  protocol: 'http',
+ *  hostname: 'domain.com',
+ *  port: 80,
+ *  pathnamePrefix?: '/v2/customers',
+ *  clientId?: 'my-client-id',
+ *  secret?: 'secret key if exists',
+ *  agent?: string
+ * }
+ */
+export class PurchasesClient extends BaseClient implements IPurchasesClient {
+
+  /**
+   * constructor
+   * @param options ClientOptions
+   */
+  constructor(options: ClientOptions) {
+    super(`${options.protocol}://${options.hostname}:${options.port}${options.pathnamePrefix}`, options);
+
+  }
+
+  /**
+   * Get customer purchases
+   * @param applicationUsername : string application-username
+   * @param cb : callback method for getting the data.
+   * @returns : void
+   */
+  getCustomerPurchases(applicationUsername: string, cb: GetCustomerPurchasesCallback): void {
+
+    if (applicationUsername === null || applicationUsername === '' || applicationUsername === undefined) {
+      return cb(new Error("UserId parameter is required to be non-null and non-empty"));
+    }
+
+    this.apiCall({
+      method: 'get', path: `/${applicationUsername}/purchases`,
+      qs: null, headers: {}, body: null
+    }, (err, result) => {
+      if (err) {
+
+        return cb(err);
+      }
+
+      cb(null, result as ApiCustomerPurchases);
+
+    });
+  }
+
+  /**
+   * Get bulk information about your customers.
+   * @param params : GetCustomersBulkInfoParams => limit, skip, applicationUsername
+   * @param cb : callback method for getting the data.
+   * @returns : void
```

Comment:
There should be an empty line after the short description.

You shouldn't add a `:` (column) here but an `-` (hyphen). If not correctly done, softwares that analyze TSDoc comments will produce weird results.

<img width="787" alt="Screen Shot 2022-01-24 at 12 03 32" src="https://user-images.githubusercontent.com/191881/150762325-5fdd5e60-5993-4d5b-a330-cf9cfbbe2ca4.png">

See https://tsdoc.org/pages/tags/param/ for the official documentation including examples.

Once cleaned up to
```ts
  /**
   * Get customer purchases
   * 
   * @param applicationUsername - The user id as registered during receipt validation.
   * @param cb - Callback method for getting the data.
   */
```

The output is nicer:

<img width="1312" alt="Screen Shot 2022-01-24 at 12 10 58" src="https://user-images.githubusercontent.com/191881/150763702-816f76db-e889-4a4e-b405-6a2998b25ddf.png">